source/s3_auto_ranged_put.c

```diff
@@ -936,15 +942,15 @@ static void s_skip_parts_from_stream_loop(void *user_data) {
             goto on_done;
         }
 
-        struct aws_byte_buf checksum_buf;
+        struct aws_byte_buf *checksum_buf;
         aws_array_list_get_at(&auto_ranged_put->synced_data.encoded_checksum_list, &checksum_buf, skip_job->part_index);
 
         // compare skipped checksum to previously uploaded checksum
-        if (checksum_buf.len > 0 && s_verify_part_matches_checksum(
-                                        meta_request->allocator,
-                                        *temp_body_buf,
-                                        meta_request->checksum_config.checksum_algorithm,
-                                        checksum_buf)) {
+        if (checksum_buf && s_verify_part_matches_checksum(
```

Comment:
this code makes me realize that the array-list entries could be in 3 states:
- NULL because ListParts says they didn't have a checksums before
- Allocated (but uninitialized) because we've read the part, but haven't calculated a checksum yet (or we're not doing checksums at all)
- Allocated, initialized, and filled

Is this an issue? That there are 2 "invalid" states (NULL or len==0) now instead of just 1?